components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -424,6 +428,27 @@ private void storeAccessToken(String accessToken, String consumerKey, AccessToke
 
     }
 
+    private boolean isPostgreSQLDB(String accessToken, String consumerKey, AccessTokenDO
+            accessTokenDO, String userStoreDomain, int retryAttempt) throws IdentityOAuth2Exception {
+
+        boolean isPostgreSQL = false;
+        Connection pgConnection = IdentityDatabaseUtil.getDBConnection();
+        try {
+            if (StringUtils.containsIgnoreCase(pgConnection.getMetaData().getDriverName(), "PostgreSQL")) {
+                recoverFromConAppKeyConstraintViolation(accessToken, consumerKey, accessTokenDO,
+                        pgConnection, userStoreDomain, retryAttempt + 1);
+                isPostgreSQL = true;
+            }
+        } catch (SQLException ex) {
+            return isPostgreSQL;
+        } finally {
+            if (isPostgreSQL) {
```

Comment:
Do we need to check isPostgreSQL while closing the connection? No matter isPostgreSQL is true/false, we have created the connection at L435. Shall we recheck the logic please?